src/std/net/socket/server-test.ss

```diff
@@ -0,0 +1,75 @@
+;;; -*- Gerbil -*-
+;;; (C) vyzo at hackzen.org
+;;; Socket server tests
+
+(import :gerbil/gambit/threads
+        :std/misc/threads
+        :std/net/socket
+        :std/net/socket/base
+        :std/os/socket
+        :std/error
+        :std/sugar
+        :std/test)
+(export socket-server-test)
+
+(def socket-server-test
+  (test-suite "test :std/net/socket"
+
+    (def server-address "localhost:4242")
+
+    (def (echo sock)
+      (def buf (make-u8vector 256))
+      (try
+       (let lp ()
+         (def len (ssocket-recv sock buf))
+         (ssocket-send sock buf 0 len)
+         (lp))
+       (catch (io-error? exn)
+         (values))))
+
+    (def (run-echo-server address)
+      (def sock (ssocket-listen address))
+      (let lp ()
+        (spawn/group 'echo-server echo (ssocket-accept sock))
+        (lp)))
+
+    (def (start-echo-server! address)
+      (parameterize ((current-socket-server #f))
+        (start-socket-server!)
+        (let (echo-server (spawn/group 'echo-server run-echo-server address))
+          (values (current-socket-server) echo-server))))
+
+    (def (stop-echo-server! server)
+      (let (tgroup (thread-thread-group server))
+        (thread-group-kill! tgroup)))
+
+    (def (receive-data sock)
+      (def buf (make-u8vector 512))
+      (def len (ssocket-recv sock buf))
+      (bytes->string (subu8vector buf 0 len)))
+
+    (define-values (socket-server echo-server)
+      (start-echo-server! server-address))
+
+    (thread-sleep! 0.2)
+
+    (test-case "test socket-server"
+      (def sock (ssocket-connect server-address))
+      (ssocket-send sock (string->bytes "Hello"))
+      (check (receive-data sock) => "Hello")
+      (ssocket-close sock))
```

Comment:
let's write and read some more data!

Reply:
Sounds good, put in a couple more additional sends.